CondFormats/SiPixelTransient/src/SiPixelTemplate.cc

```diff
@@ -1842,30 +1809,24 @@ bool SiPixelTemplate::interpolate(int id, float cotalpha, float cotbeta, float l
 
     // Do the spares next
 
-    chi2xavgone_ = ((1.f - xxratio) * thePixelTemp_[index_id_].entx[iyhigh][ilow].chi2xavgone +
-                    xxratio * thePixelTemp_[index_id_].entx[iyhigh][ihigh].chi2xavgone);
-    if (thePixelTemp_[index_id_].entx[iyhigh][imidy].chi2xavgone != 0.f) {
-      chi2xavgone_ = chi2xavgone_ / thePixelTemp_[index_id_].entx[iyhigh][imidy].chi2xavgone * chi2xavgone;
+    chi2xavgone_ = ((1.f - xxratio) * entx20_->chi2xavgone + xxratio * entx22_->chi2xavgone);
+    if (entx21_->chi2xavgone != 0.f) {
+      chi2xavgone_ = chi2xavgone_ / entx21_->chi2xavgone * chi2xavgone;
     }
 
-    chi2xminone_ = ((1.f - xxratio) * thePixelTemp_[index_id_].entx[iyhigh][ilow].chi2xminone +
-                    xxratio * thePixelTemp_[index_id_].entx[iyhigh][ihigh].chi2xminone);
-    if (thePixelTemp_[index_id_].entx[iyhigh][imidy].chi2xminone != 0.f) {
-      chi2xminone_ = chi2xminone_ / thePixelTemp_[index_id_].entx[iyhigh][imidy].chi2xminone * chi2xminone;
+    chi2xminone_ = ((1.f - xxratio) * entx20_->chi2xminone + xxratio * entx22_->chi2xminone);
+    if (entx21_->chi2xminone != 0.f) {
+      chi2xminone_ = chi2xminone_ / entx21_->chi2xminone * chi2xminone;
     }
 
-    fracxone_ = (1.f - yxratio) * ((1.f - xxratio) * thePixelTemp_[index_id_].entx[iylow][ilow].fracxone +
-                                   xxratio * thePixelTemp_[index_id_].entx[iylow][ihigh].fracxone) +
-                yxratio * ((1.f - xxratio) * thePixelTemp_[index_id_].entx[iyhigh][ilow].fracxone +
-                           xxratio * thePixelTemp_[index_id_].entx[iyhigh][ihigh].fracxone);
-    fracxtwo_ = (1.f - yxratio) * ((1.f - xxratio) * thePixelTemp_[index_id_].entx[iylow][ilow].fracxtwo +
-                                   xxratio * thePixelTemp_[index_id_].entx[iylow][ihigh].fracxtwo) +
-                yxratio * ((1.f - xxratio) * thePixelTemp_[index_id_].entx[iyhigh][ilow].fracxtwo +
-                           xxratio * thePixelTemp_[index_id_].entx[iyhigh][ihigh].fracxtwo);
+    fracxone_ = (1.f - yxratio) * ((1.f - xxratio) * entx00_->fracxone + xxratio * entx02_->fracxone) +
+                yxratio * ((1.f - xxratio) * entx20_->fracxone + xxratio * entx22_->fracxone);
+    fracxtwo_ = (1.f - yxratio) * ((1.f - xxratio) * entx00_->fracxtwo + xxratio * entx02_->fracxtwo) +
+                yxratio * ((1.f - xxratio) * entx20_->fracxtwo + xxratio * entx22_->fracxtwo);
 
     //       for(i=0; i<10; ++i) {
-    //	      pxspare[i]=(1.f - yxratio)*((1.f - xxratio)*thePixelTemp_[index_id_].entx[iylow][ilow].xspare[i] + xxratio*thePixelTemp_[index_id_].entx[iylow][ihigh].xspare[i])
-    //		          +yxratio*((1.f - xxratio)*thePixelTemp_[index_id_].entx[iyhigh][ilow].xspare[i] + xxratio*thePixelTemp_[index_id_].entx[iyhigh][ihigh].xspare[i]);
+    //	      pxspare[i]=(1.f - yxratio)*((1.f - xxratio)*entx00_->xspare[i] + xxratio*entx02_->xspare[i])
```

Comment:
Is this commented out code needed: add one line explaining its usage, if so; otherwise just remove

Reply:
added comment